static/app/views/settings/project/projectKeys/details/index.tsx

```diff
@@ -21,55 +25,58 @@ type Props = {
   {}
 >;
 
-type State = {
-  data: ProjectKey;
-} & DeprecatedAsyncView['state'];
+export default function ProjectKeyDetails({organization, params, project}: Props) {
+  const {keyId, projectId} = params;
+  const api = useApi();
+  const queryClient = useQueryClient();
 
-export default class ProjectKeyDetails extends DeprecatedAsyncView<Props, State> {
-  getTitle() {
-    return t('Key Details');
-  }
+  const {
+    data: projKeyData,
+    isLoading,
+    isFetching,
+    isError,
+  } = useApiQuery<ProjectKey>(
+    [`/projects/${organization.slug}/${projectId}/keys/${keyId}/`],
+    {staleTime: 0}
+  );
 
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization} = this.props;
-    const {keyId, projectId} = this.props.params;
-    return [['data', `/projects/${organization.slug}/${projectId}/keys/${keyId}/`]];
+  function onDataChange(data: ProjectKey) {
+    setApiQueryData<ProjectKey>(
+      queryClient,
+      [`/projects/${organization.slug}/${projectId}/keys/${keyId}/`, {}],
+      oldData => {
+        return {...oldData, data};
+      }
+    );
   }
 
-  handleRemove = () => {
-    const {organization} = this.props;
-    const {projectId} = this.props.params;
+  const handleRemove = () => {
     browserHistory.push(
       normalizeUrl(`/settings/${organization.slug}/projects/${projectId}/keys/`)
     );
   };
 
-  updateData = (data: ProjectKey) => {
-    this.setState(state => {
-      return {...state, data};
-    });
-  };
-
-  renderBody() {
-    const {organization, project, params} = this.props;
-    const {data} = this.state;
-
-    return (
-      <div data-test-id="key-details">
-        <SettingsPageHeader title={t('Key Details')} />
-        <PermissionAlert project={project} />
-
-        <KeyStats api={this.api} organization={organization} params={params} />
+  if (isError) {
+    return <RouteError />;
+  }
 
-        <KeySettings
-          data={data}
-          updateData={this.updateData}
-          onRemove={this.handleRemove}
-          organization={organization}
-          project={project}
-          params={params}
-        />
-      </div>
-    );
+  if (isLoading || isFetching) {
```

Comment:
There shouldn't be a need for `isFetching` here, including it will prevent this component from showing cached content which we want since it makes things feel faster.